vortex-ipc/src/messages/reader_sync.rs

```diff
@@ -0,0 +1,45 @@
+use std::io::Read;
+
+use bytes::BytesMut;
+use vortex_error::VortexResult;
+
+use crate::messages::{DecoderMessage, MessageDecoder, NextMessage};
+
+pub struct SyncMessageReader<R> {
```

Comment:
shall we impl `Iterator<DecoderMessage>` to mirror the `Stream<DecoderMessage>` that we impl for the async variant?